packages/gatsby-source-drupal/src/gatsby-node.js

```diff
@@ -659,14 +653,17 @@ ${JSON.stringify(webhookBody, null, 4)}`
   createNodesSpan.setTag(`sourceNodes.createNodes.count`, nodes.size)
 
   // second pass - handle relationships and back references
-  nodes.forEach(node => {
-    handleReferences(node, {
-      getNode: nodes.get.bind(nodes),
-      mutateNode: true,
-      createNodeId,
-      entityReferenceRevisions,
+  await Promise.all(
+    Array.from(nodes.values()).map(node => {
+      handleReferences(node, {
```

Comment:
I think you need to return the handleReferences promise here if you want to Promise.all it

Reply:
🤦